guestbook/backend/consul_redis.go

```diff
@@ -0,0 +1,70 @@
+package backend
+
+import (
+	"fmt"
+	"github.com/hashicorp/consul/api"
+)
+
+type ConsulRedisBackend struct {
+	redisServiceName string
+}
+
+func NewConsulRedisBackend(redisServiceName string) *ConsulRedisBackend {
+	return &ConsulRedisBackend{
+		redisServiceName: redisServiceName,
+	}
+}
+
+func (c *ConsulRedisBackend) getRedisAddress() (string, error) {
+	client, err := api.NewClient(api.DefaultConfig())
+	if err != nil {
+		return "", fmt.Errorf("Failed to connect to consul: %v", err)
+	}
+
+	services, _, err := client.Catalog().Service(c.redisServiceName, "", nil)
```

Comment:
I still have some things to iron out in my understanding of Go:

The signature of `Service()` says it takes three arguments which are `string`, `string`, and `*QueryOptions`. For that third argument, I would have created an empty `&QueryOptions{}` instead of passing `nil`. Is one approach better than the other?